bindings/node/src/mongocrypt.cc

```diff
@@ -0,0 +1,689 @@
+    return std::string(data, data + len);
+}
+
+mongocrypt_binary_t* BufferToBinary(v8::Local<v8::Object> node_buffer) {
+    uint8_t* buffer = (uint8_t*)node::Buffer::Data(node_buffer);
+    size_t buffer_len = node::Buffer::Length(node_buffer);
+
+    uint8_t* buffer_copy = new uint8_t[buffer_len];
+    memcpy(buffer_copy, buffer, buffer_len);
+
+    return mongocrypt_binary_new_from_data(buffer_copy, buffer_len);
+}
+
+v8::Local<v8::Object> BufferFromBinary(mongocrypt_binary_t* binary) {
+    Nan::EscapableHandleScope scope;
+    const uint8_t* data = mongocrypt_binary_data(binary);
+    size_t len = mongocrypt_binary_len(binary);
+
+    uint8_t* data_copy = new uint8_t[len];
+    memcpy(data_copy, data, len);
+
+    v8::Local<v8::Object> buffer = Nan::NewBuffer((char*)data_copy, len).ToLocalChecked();
+    return scope.Escape(buffer);
+}
+
+NAN_INLINE bool BooleanOptionValue(v8::Local<v8::Object> options,
+                                   const char* _key,
+                                   bool def = false) {
+    Nan::HandleScope scope;
+    v8::Local<v8::String> key = Nan::New(_key).ToLocalChecked();
+    if (options.IsEmpty() || !Nan::Has(options, key).FromMaybe(false)) {
+        return def;
+    }
+
+    v8::Local<v8::Value> value = Nan::Get(options, key).ToLocalChecked();
+    if (!value->IsBoolean()) {
+        return def;
+    }
+
+    return Nan::To<bool>(value).FromMaybe(def);
+}
+
+NAN_INLINE std::string StringOptionValue(v8::Local<v8::Object> options, const char* _key) {
+    Nan::HandleScope scope;
+    v8::Local<v8::String> key = Nan::New(_key).ToLocalChecked();
+    if (options.IsEmpty() || !Nan::Has(options, key).FromMaybe(false)) {
+        return std::string();
+    }
+
+    v8::Local<v8::Value> value = Nan::Get(options, key).ToLocalChecked();
+    if (!value->IsString()) {
+        return std::string();
+    }
+
+    return std::string(*(Nan::Utf8String(value)));
+}
+
+std::string errorStringFromStatus(mongocrypt_t* crypt) {
+    std::unique_ptr<mongocrypt_status_t, MongoCryptStatusDeleter> status(mongocrypt_status_new());
+    mongocrypt_status(crypt, status.get());
+    const char* errorMessage = mongocrypt_status_message(status.get(), NULL);
+    if (!errorMessage) {
+        return "Operation failed";
+    }
+
+    return errorMessage;
+}
+
+std::string errorStringFromStatus(mongocrypt_ctx_t* context) {
+    std::unique_ptr<mongocrypt_status_t, MongoCryptStatusDeleter> status(mongocrypt_status_new());
+    mongocrypt_ctx_status(context, status.get());
+    const char* errorMessage = mongocrypt_status_message(status.get(), NULL);
+    if (!errorMessage) {
+        return "Operation failed";
+    }
+
+    return errorMessage;
+}
+
+std::pair<bool, std::string> setKmsProviderOptions(mongocrypt_t* crypt,
+                                                   v8::Local<v8::Object> options) {
+    Nan::HandleScope scope;
+    v8::Local<v8::String> AWS_KEY = Nan::New("aws").ToLocalChecked();
```

Comment:
q: these look relatively static. Can they be defined as global consts, or does v8/nan require to be defined inside the function invocation?